client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -249,6 +284,100 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
         batchHandleChangePartitionRequestInterval,
         TimeUnit.MILLISECONDS)
     }
+
+    batchHandleCommitPartitionSchedulerThread.foreach {
+      _.scheduleAtFixedRate(
+        new Runnable {
+          override def run(): Unit = {
+            committedPartitionInfo.asScala.foreach { case (shuffleId, shuffleCommittedInfo) =>
+              batchHandleCommitPartitionExecutors.submit {
+                new Runnable {
+                  override def run(): Unit = {
+                    if (inProcessStageEndShuffleSet.contains(shuffleId) ||
```

Comment:
suggest move the if code block into shuffleCommittedInfo synchronized block in the else branch，because It may be overlap as stageEnd commit and this commit happen at the same time，commitPartitionRequests may contains partitions which are already committed by stageEnd process。